src/tasks/block.rs

```diff
@@ -172,6 +187,54 @@ impl BlockBuilder {
         self.bundle_poller.evict();
     }
 
+    /// Simulates a Flashbots-style ZenithEthBundle, simualating each transaction in it's bundle
+    /// by calling it against the host provider at the current height against default storage (no state overrides)
+    /// and failing the whole bundle if any transaction not listed in the reverts list fails that call.
+    async fn simulate_bundle(
+        &mut self,
+        bundle: &ZenithEthBundle,
+        host_provider: &Provider,
+    ) -> eyre::Result<()> {
+        tracing::info!("simulating bundle");
+
+        let reverts = &bundle.bundle.reverting_tx_hashes;
+        tracing::debug!(reverts = ?reverts, "processing bundle with reverts");
+
+        for tx in &bundle.bundle.txs {
+            let (tx_env, hash) = self.parse_from_bundle(tx)?;
+
+            // Simulate and check for reversion allowance
+            match self.simulate_transaction(host_provider, tx_env).await {
+                Ok(_) => {
+                    // Passed, log trace and continue
+                    tracing::debug!(tx = %hash, "tx passed simulation");
+                    continue;
+                }
+                Err(sim_err) => {
+                    // Failed, only continfue if tx is marked in revert list
+                    tracing::debug!("tx failed simulation: {}", sim_err);
+                    if reverts.contains(&hash) {
+                        continue;
+                    } else {
+                        bail!("tx {hash} failed simulation but was not marked as allowed to revert")
+                    }
+                }
+            }
+        }
+        Ok(())
+    }
+
+    /// Simulates a transaction by calling it on the host provider at the current height with the current state.
+    async fn simulate_transaction(
+        &self,
+        host_provider: &Provider,
+        tx_env: TxEnvelope,
+    ) -> eyre::Result<()> {
+        let tx = TransactionRequest::from_transaction(tx_env);
+        host_provider.call(&tx).await?;
```

Comment:
❗ problem: isn't this a bundle of rollup transactions? this is simulating them against the host provider.